test/form_builder/input_field_test.rb

```diff
@@ -173,4 +173,29 @@ def with_input_field_for(object, *args)
       assert_select 'input[readonly="readonly"]'
     end
   end
+
+  test 'adds valid class to input_field when it is configured' do
+    @user.instance_eval { undef errors }
+    SimpleForm.input_field_valid_class = 'is-valid'
```

Comment:
WDYT to use [`swap`](https://github.com/plataformatec/simple_form/blob/dd05b56df12c7a21368f7870b1cff92abc793656/test/support/misc_helpers.rb#L15) here to ensure the state is cleaned after the test runs?
```ruby
swap SimpleForm, input_field_valid_class: 'is-valid' do
  @user.instance_eval { undef errors }
  with_input_field_for @user, :name
  assert_select 'input.string.required.is-valid'
end

Reply:
Good catch 👍